### PR TITLE
[MLIR]Add clang flag and skip bf16 tests for MLIR.

### PR DESCRIPTION
**Pull request by ayzhuang**

### CODE REVIEW

File: cmake/external_llvm.cmake [hunk 1]
```diff
@@ -91,6 +91,7 @@ ExternalProject_Add(
                 -DLIBOMP_OMPT_SUPPORT=OFF
                 -DCLANG_BUILD_TOOLS=OFF
                 -DCLANG_ENABLE_ARCMT=OFF
+                -DCLANG_ANALYZER_ENABLE_Z3_SOLVER=OFF
```
Comment (silee2) [inline cmake/external_llvm.cmake]: Why is this change related to MLIR? This LLVM is the one used for CPU codegen.

Comment (ayzhuang) [reply cmake/external_llvm.cmake]: For CI we build LLVM only when MLIR is enabled. This flag is needed when building LLVM8 with clang9 or clang10.

File: test/cpu_test.cpp [hunk 2]
```diff
@@ -2157,10 +2157,10 @@ TEST(cpu_test, tensor_copy_from_different_layout)
 
 TEST(cpu_test, max_pool_bf16)
 {
-    if (!runtime::cpu::mkldnn_utils::is_bf16_supported())
+    if (!runtime::cpu::mkldnn_utils::is_bf16_supported() || std::getenv("NGRAPH_MLIR") != nullptr)
```
Comment (diyessi) [inline test/cpu_test.cpp]: Why not just compile in the disabling of the test?

Comment (ayzhuang) [reply test/cpu_test.cpp]: You mean to put #ifndef outside to not compile this test when MLIR is enabled?
